### PR TITLE
Fixes #302.

### LINKED ISSUE #302: Matplotlib example not working
Most likelly is a problem with my setup, but when running that example 
[Example](https://jeltef.github.io/PyLaTeX/current/examples/matplotlib_ex.html)

Result:
`LaTeX Warning: File `C:/Users/User/AppData/Local/Temp/{pylatex-tmp.45dxkgof/5131bfa5-603e-4fee-9366-f8aa72977fb0}.pdf' not found `

It seems as if matplotlib is supposed to generate a pdf with the chart to later be integrated in the document, but it is never created. How could I troubleshoot it?

### PR DESCRIPTION
**Pull request by zthurman**
@JelteF Testall.sh passed on my Windows 10 and Manjaro both. Let me know how it looks on your side

### CODE REVIEW

File: pylatex/utils.py [hunk 1]
```diff
@@ -131,7 +136,10 @@ def fix_filename(path):
     file_parts = filename.split('.')
 
     if len(file_parts) > 2:
-        filename = '{' + '.'.join(file_parts[0:-1]) + '}.' + file_parts[-1]
+        if os.name == 'posix':
+            filename = '{' + '.'.join(file_parts[0:-1]) + '}.' + file_parts[-1]
+        else:
+            filename = '.'.join(file_parts[0:-1]) + '.' + file_parts[-1]
```
Comment (JelteF) [inline pylatex/utils.py:142]: This seems like a very convoluted way to make the filename end up in the same way that it already was. I think it makes more sense to change this:
```python
len(file_parts) > 2
```
To:
```python
os.name == 'posix' and len(file_parts) > 2
```

And keep the rest at it originally was (no extra `else` block)

File: pylatex/utils.py [hunk 2]
```diff
@@ -102,10 +102,15 @@ def fix_filename(path):
     Latex has problems if there are one or more points in the filename, thus
     'abc.def.jpg' will be changed to '{abc.def}.jpg'
 
+    Windows gets angry about the brackets that resolve the above issue on linux
```
Comment (JelteF) [inline pylatex/utils.py:105]:
```suggestion
    Windows gets angry about the curly braces that resolve the above issue on linux
```

File: tests/test_utils_fix_filename.py [hunk 3]
```diff
@@ -15,7 +16,10 @@ def test_one_dot():
 
 def test_two_dots():
     fname = "aa.a.a"
-    assert fix_filename(fname) == "{aa.a}.a"
+    if os.name == 'posix':
```
Comment (JelteF) [inline tests/test_utils_fix_filename.py:19]: Is it possible to test both, by manually changing `os.name`?
Something like this:
```python
original_os_name = os.name
try:
    os.name = 'posix'
    assert fix_filename(fname) == "{aa.a}.a"
    os.name = 'nt'
    assert fix_filename(fname) == "aa.a.a"
finally:
    os.name = original_os_name
```

**Review by JelteF — CHANGES_REQUESTED** [review]
Thanks for the contribution! Left some small comments.

**Review by zthurman — COMMENTED** [review]
Added requested changes, example and tests still working on both linux and Windows.

**Comment by JelteF** [comment]
This is merged and released as 1.4.1

**Pull request closed without merging**